controllers/powerworkload_controller.go

```diff
@@ -493,13 +562,61 @@ func (r *PowerWorkloadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// shouldEnqueueWorkloadForNode returns true if this node-agent should reconcile the workload.
+func (r *PowerWorkloadReconciler) shouldEnqueueWorkloadForNode(obj client.Object) bool {
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		return false
+	}
+
+	workload, ok := obj.(*powerv1.PowerWorkload)
+	if !ok {
+		return false
+	}
+
+	// Exclusive workloads are named <profile>-<nodeName>.
+	if !workload.Spec.AllCores {
+		return strings.HasSuffix(workload.Name, "-"+nodeName)
+	}
+
+	// If the shared workload is already claimed by a specific node, only enqueue on that node.
+	if workload.Status.WorkloadNodes.Name != "" {
+		return workload.Status.WorkloadNodes.Name == nodeName
+	}
+
+	// Shared workloads supposedly have a node selector associated with them.
+	// If no selector is provided in the shared workload, we can't identify the node it is associated with.
+	if len(workload.Spec.PowerNodeSelector) == 0 {
+		r.Log.Info("No node selector provided for shared workload, ignoring", "workload", workload.Name)
+		return false
+	}
+
+	// Otherwise, check if the current node matches the label selector map.
```

Comment:
I'm wondering in which situation we'd make it this far here 🤔 When the PowerWorkload does not specify a PowerNodeSelector AND its status hasn't been updated it to include the node name?

Reply:
> When the PowerWorkload does not specify a PowerNodeSelector AND its status hasn't been updated it to include the node name?

right, two situations where this could happen:
- shared workload initially created with an empty selector
- when the selector is cleared after creation, the original owner clears workload.Status.WorkloadNodes.Name, triggering the event